genie-agent/src/main/java/com/netflix/genie/agent/execution/services/ArchivalService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import com.netflix.genie.agent.execution.exceptions.ArchivalException;
+
+import java.io.File;
+import java.net.URI;
+
+/**
+ * Service to archive a local file to a target location URI.
```

Comment:
Description makes it sound like it uploads a single file.

Reply:
Maybe this service should explicitly require the first argument to be a folder?
I know the current implementation works for both regular and directory, but would make it simpler to say it archives a directory recursively to a remote location.

Reply:
Will fix the comment